src/Tools/dotnet-openapi/src/Commands/RefreshCommand.cs

```diff
@@ -0,0 +1,73 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using Microsoft.Extensions.CommandLineUtils;
+using Microsoft.Extensions.Tools.Internal;
+
+namespace Microsoft.DotNet.OpenApi.Commands
+{
+    internal class RefreshCommand : BaseCommand
+    {
+        private const string CommandName = "refresh";
+
+        public RefreshCommand(Application parent) : base(parent, CommandName)
+        {
+            _sourceFileArg = Argument(SourceProjectArgName, $"The openapi reference to refresh.");
+        }
+
+        internal readonly CommandArgument _sourceFileArg;
+
+        protected override async Task<int> ExecuteCoreAsync()
+        {
+            var projectFile = ResolveProjectFile(ProjectFileOption);
+
+            var sourceFile = Ensure.NotNullOrEmpty(_sourceFileArg.Value, SourceProjectArgName);
+
+            if (IsUrl(sourceFile))
+            {
+                var destination = FindReferenceFromUrl(projectFile, sourceFile);
+                using (var client = new HttpClient())
```

Comment:
FYI HttpClient is designed to be reused instead of being instantiated and disposed. However, since the commandline tool isn't long lived, it's not a problem here.

Reply:
Good to bring it up though, I've definitely been guilty of throwing them away too much before, so reminders never hurt.

Reply:
Also, turns out we didn't actually use that client.